railgun/pkg/config/config.go

```diff
@@ -197,3 +207,42 @@ func (c *Config) GetKubeClient() (client.Client, error) {
 	}
 	return client.New(conf, client.Options{})
 }
+
+func (c *Config) ConfigKongService(ctx context.Context) error {
+	// deployment configuration
+	adminApiService := os.Getenv("CONTROLLER_KONG_ADMIN_PUBLISH_SERVICE")
+	if adminApiService == "" {
+		// parameter configuration
+		if len(c.KongAdminURL) > 0 && c.KongAdminURL != "http://localhost:8001" {
+			return nil
+		}
+		return fmt.Errorf(" kong admin api url %s should not be localhost.", c.KongAdminURL)
+	}
+
+	kubeCfg, err := c.GetKubeconfig()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve kubeconfig. err %v", err)
+	}
+
+	if adminApiService == "" {
+		// parameter configuration
+		if len(c.KongAdminAPI) > 0 {
+			adminApiService = c.KongAdminAPI
+		} else {
+			return fmt.Errorf("could not find kong admin api service namespace and name information.")
+		}
+	}
+	kongadminurl, err := ctrlutils.RetrieveKongAdminAPIURL(ctx, adminApiService, kubeCfg)
+	if err != nil || len(kongadminurl) == 0 {
+		return fmt.Errorf("failed to generating kong admin url. err %v", err)
+	}
+	c.KongAdminURL = kongadminurl
+
+	if len(c.PublishService) == 0 {
+		kongService := os.Getenv("CONTROLLER_PUBLISH_SERVICE")
```

Comment:
question: are we setting a _new precedent_ here that the flags will override ENV vars?

Would it potentially be better for us to pre-emptively parse ENV vars and configuring the *config.Config option accordingly in a single place that's meant for this purpose? :thinking:

Reply:
We don't overwrite env vars. On the other way, we read env vars and use it configure the kong Config options.